src/tree.c

```diff
@@ -135,7 +135,7 @@ static git_tree_entry *alloc_entry(const char *filename)
 
 struct tree_key_search {
 	const char *filename;
-	uint16_t filename_len;
+	size_t filename_len;
```

Comment:
Hm, yes that's true.  A quick perusal of the git source shows that they use an [`unsigned int`](https://github.com/git/git/blob/563e38491eaee6e02643a22c9503d4f774d6c5be/tree-walk.h#L13) here for the tree entry length.

There's an obvious compatibility issue here, then.  And although I'm not surprised that we haven't run into it yet, one can certainly imagine a world where somebody is using git as a general data exchange mechanism (perhaps not directly related to filesystem or filename sizes) and a `uint16_t` is too small to hold the tree entries.  So we should probably increase `git_tree_entry`'s size as well.